sqlglot/dialects/redshift.py

```diff
@@ -388,13 +389,28 @@ def unnest_sql(self, expression: exp.Unnest) -> str:
             args = expression.expressions
             num_args = len(args)
 
-            if num_args > 1:
+            if not num_args == 1:
                 self.unsupported(f"Unsupported number of arguments in UNNEST: {num_args}")
                 return ""
 
-            arg = self.sql(seq_get(args, 0))
+            if not expression.find_ancestor(exp.From, exp.Join):
+                self.unsupported("Unsupported UNNEST when not used in FROM/JOIN clauses")
+                return ""
```

Comment:
This is gonna traverse all the way up, so if you have a derived table you may hit the parent scope's `From` or `Join`. I think we want to do include `Select` and then check the ancestor is not an `Select`.